frontend2/src/components/team/EligibilitySettings.tsx

```diff
@@ -100,18 +113,44 @@ const EligibilitySettings: React.FC = () => {
                     desiredEligibility.includes(criterion.id),
                     (team.profile?.eligible_for ?? []).includes(criterion.id),
                   )}
-                  onChange={(checked) => {
+                  onChange={async (checked) => {
                     if (checked && !desiredEligibility.includes(criterion.id)) {
-                      setDesiredEligibility([
-                        ...desiredEligibility,
-                        criterion.id,
-                      ]);
+                      // Add criterion to team
+                      // We do NOT want to await this, because we want to send the state of this mutation to isPending!
+                      updateTeam({
+                        episodeId,
+                        patchedTeamPrivateRequest: {
+                          profile: {
+                            ...team.profile,
```

Comment:
passing team.profile in here results in race conditions with the rest of the page (e.g. if ScrimmageSettings or MyTeam attempts to also update the profile, but their changes get overriden by this call)
we shouldn't need to pass team.profile here b/c it is a partial update